numpy/_core/src/multiarray/stringdtype/casts.c

```diff
@@ -1660,6 +1909,20 @@ get_casts()
     DTYPES_AND_CAST_SPEC(cdouble, CDouble)
     DTYPES_AND_CAST_SPEC(clongdouble, CLongDouble)
 
+    PyArray_DTypeMeta **s2v_dtypes = get_dtypes(
+            (PyArray_DTypeMeta *)&StringDType, &PyArray_VoidDType);
+
+    PyArrayMethod_Spec *StringToVoidCastSpec = get_cast_spec(
+            s2v_name, NPY_UNSAFE_CASTING, NPY_METH_NO_FLOATINGPOINT_ERRORS,
+            s2v_dtypes, s2v_slots);
+
+    PyArray_DTypeMeta **v2s_dtypes = get_dtypes(
+            &PyArray_VoidDType, (PyArray_DTypeMeta *)&StringDType);
+
+    PyArrayMethod_Spec *VoidToStringCastSpec = get_cast_spec(
+            v2s_name, NPY_SAFE_CASTING, NPY_METH_NO_FLOATINGPOINT_ERRORS,
```

Comment:
Are all void casts safe?  Safe sounds strange to me...

Reply:
It's safe in the sense that if the void buffer contains valid UTF-8, it will be faithfully reconstructed as a string. I don't think it matters much what the safety is, I'll make it unsafe to match the other way since there's a risk that round-tripping won't work unless the user selects a wide enough void dtype.